### PR TITLE
Dos normalization

### PR DESCRIPTION
**Pull request by ndaelman-hu**
This branch resolves the issues https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/issues/868 and https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/issues/728 by (1) correcting the scaling constants and (2) depending on the version scaling the DOS (not the projected DOS or bands) by a factor of 2.

It moreover adds tests for the changing Fermi level format.

### CODE REVIEW

File: tests/test_fhiaimsparser.py [hunk 1]
```diff
@@ -117,59 +120,67 @@ def test_band_spinpol(parser):
...
-    # reference energy
-    energies = energies.flatten()
-    energies.sort()
-    lowest_unoccupied_index = np.searchsorted(energies, energy_reference, "right")
-    highest_occupied_index = lowest_unoccupied_index - 1
-    gap = energies[lowest_unoccupied_index] - energies[highest_occupied_index]
-    assert gap == approx(0.60684)
-
-
-def test_dos_silicon(silicon):
```
Comment (ladinesa) [inline tests/test_fhiaimsparser.py:159]: You can try using pytest.parameterize here so you get rid of the for loop and you can specify which version to test.

File: tests/test_fhiaimsparser.py [hunk 2]
```diff
@@ -117,59 +120,67 @@ def test_band_spinpol(parser):
...
-    # Check that an approporiately sized band gap is found at the given
-    # reference energy
-    nonzero = np.unique(values.nonzero())
-    energies = energies[nonzero]
-    energies.sort()
-    lowest_unoccupied_index = np.searchsorted(energies, energy_reference, "right")
-    highest_occupied_index = lowest_unoccupied_index - 1
-    gap = energies[lowest_unoccupied_index] - energies[highest_occupied_index]
-    assert gap == approx(0.54054054)
+    for silicon in silicons.values():
```
Comment (ladinesa) [inline tests/test_fhiaimsparser.py]: see previous comment.

**Comment by ladinesa** [comment]
Can you also address the linting fix in exciting parser and resolve the conflict for fhi aims?

File: tests/test_fhiaimsparser.py (hunk 1 above)
Comment (ndaelman-hu) [reply tests/test_fhiaimsparser.py:151]: Sure, though both have to be tested really

Comment (ndaelman-hu) [reply tests/test_fhiaimsparser.py:151]: Sure, but both files have to be tested check the fermi level versions

**Comment by ndaelman-hu** [comment]
> address the linting fix in exciting parser
What do you mean with address? Resolving or adding documentation?

**Comment by ladinesa** [comment]
> > address the linting fix in exciting parser
> > What do you mean with address? Resolving or adding documentation?

electronicparsers/exciting/parser.py:1183:8: W0612: Unused variable 'volume' (unused-variable)

**Comment by ndaelman-hu** [comment]
Done

**Review by ladinesa — APPROVED** [review]

**Comment by ladinesa** [comment]
Please merge this.

**Comment by ndaelman-hu** [comment]
As discussed in another merge request (https://gitlab.mpcdf.mpg.de/nomad-lab/nomad-FAIR/-/merge_requests/945), I moved this quantity to the parser. That's also why I held off on merging.

**Review by ladinesa — APPROVED** [review]

**Pull request merged**